core/renderers/common/info.js

```diff
@@ -472,6 +487,11 @@ Blockly.blockRendering.RenderInfo.prototype.getInRowSpacing_ = function(prev, ne
       return next.notchOffset - this.constants_.CORNER_RADIUS;
     }
   }
+  
+  // Spacing between a dummy input and anything.
+  if (prev && Blockly.blockRendering.Types.isDummyInput(prev)) {
```

Comment:
This will still add spacing after a dummy input and before the next thing, which will come up in an inline render of the blocks you tested.